src/os/bluestore/BlueStore.cc

```diff
@@ -3661,10 +3672,26 @@ void BlueStore::ExtentMap::reshard(
     }
     dout(30) << " extent " << *extent << dendl;
 
-    // disfavor shard boundaries that span a blob
-    bool would_span = (extent->logical_offset < max_blob_end) || extent->blob_offset;
-    if (estimate &&
-	estimate + extent_avg > target + (would_span ? slop : 0)) {
+
+    bool make_shard_here = false;
+    if (onode_data_has_boundaries) {
+      if (extent->blob_start() >= next_boundary) {
+	// this it the place we want to have shard boundary
+	if ((estimate >= target /*we have enough already*/) ||
+	    (estimate + encoded_segment_estimate >= target * 3 / 2)
```

Comment:
why the 3/2?

Reply:
We have a situation when:
estimate < target
which means that we would like to pack to the shard something more.
But, if we pack one more element, we will be more than
target + 0.5 * target.

The decision is taken to accept that the current shard will be smaller.
Please note comment "we will be too large if we wait for next segment".